### PR TITLE
feat(metric): add data count

### PR DESCRIPTION
**Pull request by GrayFlash**

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -101,6 +101,7 @@ func (r *Agent) Run(recipe recipe.Recipe) (run Run) {
 		ctx         = context.Background()
 		getDuration = r.startDuration()
 		stream      = newStream()
+		dataCount   = 0
```
Comment (ravisuhag) [inline agent/agent.go:104]: @GrayFlash Let's call it `recordCount` ?

File: agent/monitor.go [hunk 2]
```diff
@@ -8,13 +8,13 @@ import (
 
 // Monitor is the interface for monitoring the agent.
 type Monitor interface {
-	RecordRun(recipe recipe.Recipe, durationInMs int, success bool)
+	RecordRun(recipe recipe.Recipe, durationInMs, docCount int, success bool)
```
Comment (ravisuhag) [inline agent/monitor.go:11]: How about we call this `recordCount`?

File: agent/agent.go (hunk 1 above)
Comment (GrayFlash) [reply agent/agent.go:104]: Sure makes sense, will add it to the report table in `run.go` as the column named `Records` right??

File: metrics/statsd.go [hunk 3]
```diff
@@ -29,35 +31,41 @@ func NewStatsdMonitor(client statsdClient, prefix string) *StatsdMonitor {
...
-		m.createMetricName(runDurationMetricName, recipe, success),
-		int64(duration),
+		m.createMetricName(runDurationMetricName, run.Recipe, run.Success, run.RecordCount),
+		int64(run.DurationInMs),
 	)
 	m.client.Increment(
-		m.createMetricName(runMetricName, recipe, success),
+		m.createMetricName(runMetricName, run.Recipe, run.Success, run.RecordCount),
+	)
+	m.client.IncrementByValue(
```
Comment (StewartJingga) [inline metrics/statsd.go:42]: i'm still thinking if we need another metric type for data count.
at first im thinking data count is just a metadata in a metric so we can put it as a tag.

let me try it out first if it we can actually get something out of this new metric.

**Pull request merged**